Yoast/yoast-components
# Add Help Center toggle button.

## Summary

This PR can be summarized in the following changelog entry:

* Add Help Center toggle button component

## Relevant technical choices:

* this button seems to me a unique case, so I've put it in its own file
* can receive an `isExpanded` prop from a wrapper component
* added `isExpanded` prop to the base YoastButton
* changed the `children` prop to accept, array of nodes, node, string

**Question:**
worth noting in the design in https://github.com/Yoast/yoast-components/issues/311 and https://github.com/Yoast/yoast-components/issues/196 this button seems to have squared corners even if the comment says "The button is of the same class as all the other My Yoast buttons". However, the other buttons have rounded corners... not a big deal, `border-radius` can be reset but needs clarification from design. /cc @hedgefield 

## Test instructions

This PR can be tested by following these steps:

* test with the following code in ContentAnalysis and check the new button matches the design
```
iimport React from "react";
import styled from "styled-components";

import { IconButton } from "../../Shared/components/Button";
import { questionCircle } from "../../../../style-guide/svg";
import { YoastButton } from "../../Shared/components/YoastButton";
import { YoastLinkButton } from "../../Shared/components/YoastLinkButton";
import { HelpCenterButton } from "../../Shared/components/HelpCenterButton";
import { makeOutboundLink } from "../../../../utils/makeOutboundLink";

export const ContentAnalysisContainer = styled.div`
	min-height: 700px;
	width: 100%;
	background-color: white;

	button {
		margin-right: 16px;
	}
`;

const RightYoastLinkButton = makeOutboundLink( styled( YoastLinkButton )`
	float: right;
` );

/**
 * Returns the ContentAnalysis component.
 *
 * @returns {ReactElement} The ContentAnalysis component.
 */
export default function ContentAnalysis() {
	return <ContentAnalysisContainer>
		<IconButton icon={ questionCircle } iconColor="#f00">Icon Button</IconButton>
		<YoastButton>Button</YoastButton>
		<HelpCenterButton onClick={ () => {
			console.log( "Toggle me!" );
		} }>Need Help?</HelpCenterButton>
		<RightYoastLinkButton href="#somewhere">Children string</RightYoastLinkButton>
		<RightYoastLinkButton href="#somewhere"><span>Children node</span></RightYoastLinkButton>
	</ContentAnalysisContainer>;
}
```

current rendering:

<img width="556" alt="screen shot 2017-09-28 at 16 11 15" src="https://user-images.githubusercontent.com/1682452/30971401-12eca01a-a468-11e7-9432-c36b8f76340e.png">


Fixes #311
